Automattic/jetpack
# Domain-only sites: Add missing menu items to the sidebar

As an oversight, menu items for the management of the `domain`, `email` and `inbox` were excluded from the domain-only sidebar while we added them to other site types.

This set of changes adds the necessary menu items that makes sense for domain-only sites, with the conditional addition of  a menu item `Manage Email`  when a site contains an active email subscription.



#### Changes proposed in this Pull Request:

* Adds menu items:
    * Manage Domain
    * Manage Email ( conditionally present when an email subscription is present )
    * Manage Purchases 
    * Inbox

#### Other information:


#### Jetpack product discussion

pbLl1t-1f6-p2#comment-1252


#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






- Checkout this branch
- Switch sites: Select a domain-only site. 
    - If you don't have one, create one using the domain-only start flow: https://wordpress.com/domains/
    - When prompted to purchase a domain, select "Just buy a domain"
- Clear your browser cache to ensure you are not working with cached items
- Confirm that you see following new menu items in the sidebar, and that they point at intuitive destinations
    - Manage Domain
    - Manage Email
        -  For a site that has no email subscriptions, confirm that this item is hidden
        -  Add an email subscription. Confirm that you now see this item
        - Remove the email subscription. Confirm that the menu item is gone after a refresh 
    - Manage Purchases
    - Inbox
- Switch sites: Select each of simple ( normal ),  atomic, etc sites and confirm that you do NOT see the menu items.

<img width="1163" alt="Screenshot 2022-09-08 at 6 07 07 AM" src="https://user-images.githubusercontent.com/277661/189040908-19f93b8d-37d2-4df7-aab8-01ed73aadfaa.png">
